posthog/models/instance_setting.py

```diff
@@ -22,8 +23,14 @@ def value(self):
 def get_instance_setting(key: str) -> Any:
     assert key in CONSTANCE_CONFIG, f"Unknown dynamic setting: {repr(key)}"
 
+    cached_setting = cache.get(key)
+
+    if cached_setting:
+        return json.loads(cached_setting)
+
     saved_setting = InstanceSetting.objects.filter(key=CONSTANCE_DATABASE_PREFIX + key).first()
     if saved_setting is not None:
+        cache.set(key, saved_setting.raw_value)
```

Comment:
This is really cool! 

The default cache is 5 minutes https://docs.djangoproject.com/en/3.2/ref/settings/#std-setting-CACHES-TIMEOUT

If we bust the cache in the `set_instance_setting` method 

https://github.com/PostHog/posthog/blob/fa19fa1f23efb275d55421389592d7e96649caf8/posthog/models/instance_setting.py#L54

then we can set a longer cache value?

(we should probably bust/update the cache in the set method anyway

Reply:
I thought about it, but I don't think the `set_instance_setting` method will be called if we update the value through the Django Admin UI (which is the use case for this configuration).

Reply:
We could probably make this work if we override the model's `save` method to clear the cache (instead of `set_instance_setting`)...

Reply:
I'm updating the cache on set as otherwise we break tests :sweat_smile:. I'm still thinking about the save idea, would mean that we never have to hit the database (assuming we bump the cache timeout as suggested).

Reply:
Would probably be frustrating to make a change in the admin UI and then have to wait up to 5 minutes to see it apply

But could probably add that extra work in a follow-up rather than add to much here :)

Reply:
Yeah, that's true. Especially considering this setting is one we'll want to mess with during outages...

At least the groundwork is here, but I'll dig a bit into how we can refresh the cache on django admin UI updates. Like I said, it will probably require overriding `save` or something similar.

Reply:
We are now setting cache on model `save` which means we work with Django Admin UI. This would also mean we could bump the cache timeout: as long as these values cannot be set directly in the database (bypassing django) then it should be safe.